lib/serializer.js

```diff
@@ -50,17 +50,14 @@ module.exports = class Serializer {
   }
 
   asNumber (i) {
-    if (typeof i !== 'number') {
-      i = Number(i)
-    }
-    // NaN !== NaN
-    if (i !== i) { // eslint-disable-line no-self-compare
+    const num = Number(i)
+    if (Number.isNaN(num)) {
       throw new Error(`The value "${i}" cannot be converted to a number.`)
+    } else if (!Number.isFinite(num)) {
```

Comment:
@Uzlopak I will approve it because I'm not if it makes any perf difference with `i === Infinity || i === -Infinity`, but if it does, feel free to open a new PR.